File: app/src/main/java/com/robotnec/budget/activity/AddAccountActivity.java
package com.robotnec.budget.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.robotnec.budget.R;
import com.robotnec.budget.core.domain.Account;
import com.robotnec.budget.fragments.AddAccountFragment;
import com.robotnec.budget.util.Keys;

/**
 * @author zak <zak@swingpulse.com>
 */
public class AddAccountActivity extends com.robotnec.budget.activity.BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            Bundle args = getIntent().getExtras();
            Account account = (Account) args.getSerializable(Keys.ACCOUNT);
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.frame_fragment, AddAccountFragment.newInstance(account))
                    .commit();
        }
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_add_account;
    }
}

File: app/src/main/java/com/robotnec/budget/app/activity/AddAccountActivity.kt
package com.robotnec.budget.app.activity

import android.os.Bundle
import com.robotnec.budget.R
import com.robotnec.budget.app.fragments.AddAccountFragment
import com.robotnec.budget.app.util.Keys
import com.robotnec.budget.app.util.newFragment
import com.robotnec.budget.core.domain.Account

/**
 * @author zak zak@swingpulse.com>
 */
class AddAccountActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            val args = intent.extras
            val account = args.getSerializable(Keys.ACCOUNT) as Account
            supportFragmentManager.beginTransaction()
                    .replace(R.id.frame_fragment, newFragment<AddAccountFragment>(AddAccountFragment.ARG_ACCOUNT to account))
                    .commit()
        }
    }

    override fun layoutId(): Int {
        return R.layout.activity_add_account
    }
}
